test/stability-tests/src/it/java/software/amazon/awssdk/stability/tests/utils/StabilityTestRunner.java

```diff
@@ -260,19 +262,24 @@ private static void processResult(TestResult testResult) {
 
         double ratio = expectedExceptionCount / (double) testResult.totalRequestCount();
         if (clientExceptionCount > 0) {
-            throw new RuntimeException(String.format("%s SdkClientExceptions were thrown, failing the tests",
-                                                     clientExceptionCount));
+            String errorMessage = String.format("%s SdkClientExceptions were thrown, failing the tests",
+                                          clientExceptionCount);
+            log.error(() -> errorMessage);
+            throw new StabilityTestsFailedException(errorMessage);
```

Comment:
Any reason we can't just throw a runtime exception to fail the test?

Reply:
No particular reason, just to match `StabilityTestsRetriableException`. I guess I can change it to `AssertionError` to make it more clear.